### PR TITLE
created webextension with an example on cookie api

### PR DESCRIPTION
**Pull request by DarKFlameS**
As suggested in the contribution MD, made an example using the cookie api.

### CODE REVIEW

**Comment by wbamberg** [comment]
Thanks a lot for this, and I'm really sorry to take so long to look at this. It's a nice example. I think it would be more interesting if it listed cookies for the active tab's document's domain, rather than being hardcoded to MDN.

That's my main comment, I have some little comments as well, which I'll add now...

File: list-cookies/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# list-cookies
+
+## What it does
+
+This extensions list the cookies in the mozilla developers domain
```
Comment (wbamberg) [inline list-cookies/README.md]: -> " This extensions lists the cookies in the developer.mozilla.org domain."

...or something like that. But also, see my main comment: https://github.com/mdn/webextensions-examples/pull/114#issuecomment-255877894.

File: list-cookies/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+# What it shows
+
+Demonstration of cookie api function cookie.getAll()
```
Comment (wbamberg) [inline list-cookies/README.md]: -> "Demonstration of the `getAll()` function in the `cookie` API."

File: list-cookies/cookies.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+
+<html>
+  <head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="cookies.css"/>
+  </head>
+
+<body>
+  <h3>Current cookies</h3>
```
Comment (wbamberg) [inline list-cookies/cookies.html]: I think the top-level heading should be an H2.

File: list-cookies/cookies.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+window.onload = function(e){
```
Comment (wbamberg) [inline list-cookies/cookies.js]: I think you don't need window.onload here, do you?

File: list-cookies/cookies.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    //get all cookies in the domain
+    chrome.cookies.getAll({domain: "developer.mozilla.org"},  function(cookies) {
+        var cookieList = document.getElementById('cookie-list');
+        //add an <li> item with the name and value of the cookie to the list
+        for (x in cookies) {
```
Comment (wbamberg) [inline list-cookies/cookies.js]: I'd rather see `for...of` than `for...in` here: https://hacks.mozilla.org/2015/04/es6-in-depth-iterators-and-the-for-of-loop/

File: list-cookies/cookies.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+window.onload = function(e){
+    //get all cookies in the domain
```
Comment (wbamberg) [inline list-cookies/cookies.js]: The convention usually followed in this repo is for 2-space indentation. Sorry, we should mention this :-/

File: list-cookies/manifest.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
+  "applications": {
+    "gecko": {
+      "id": "list-cookies@mozilla.org",
+      "strict_min_version": "47.0a1"
+    }
+  },
+  "browser_action": {
+      "default_title": "List mozilla developers domain cookies",
```
Comment (wbamberg) [inline list-cookies/manifest.json]: -> "List cookies belonging to developer.mozilla.org" or something

**Comment by DarKFlameS** [comment]
Ok, I'll solve the issues soon.

**Comment by DarKFlameS** [comment]
The problem on listing the active tab cookies is that cookies API needs host_permission. It only works on the domains listed in the manifest.json.

**Comment by wbamberg** [comment]
Have you tried it with the <all_urls> permission?

**Comment by DarKFlameS** [comment]
I did not knew the <all_urls> permission. It worked perfectly, so now, it shows the cookies of the active tab. Thanks for the advice @wbamberg

File: list-cookies/cookies.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    if (cookies.length > 0) {
+      //add an <li> item with the name and value of the cookie to the list
+      for (cookie of cookies) {
+        var li = document.createElement("li");
+        var content = document.createTextNode(cookie.name + ": "+ cookie.value);
+        li.appendChild(content);
+        cookieList.appendChild(li);
+      }
+    } else {
+      var p = document.createElement("p")
```
Comment (wbamberg) [inline list-cookies/cookies.js]: should have a semicolon

File: list-cookies/cookies.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      var content = document.createTextNode("No cookies in this tab.");
+      var parent = cookieList.parentNode
```
Comment (wbamberg) [inline list-cookies/cookies.js]: should have a semicolon

**Comment by wbamberg** [comment]
Thanks for the updates! This seems to work great. I had some minor style comments.

File: tabs-tabs-tabs/manifest.json [hunk 10]
```diff
@@ -11,6 +11,9 @@
       "default_popup": "tabs.html"
   },
   "description": "A list of methods you can perform on a tab.",
+  "icons": {
+    "48": "../icons/quicknote-48.png"
+  },
```
Comment (wbamberg) [inline tabs-tabs-tabs/manifest.json]: I think the PR didn't mean to include this change.

Comment (DarKFlameS) [inline tabs-tabs-tabs/manifest.json]: sorry, it was non-intended

**Comment by wbamberg** [comment]
Looks great. Thanks for the new example @DarKFlameS !

**Pull request merged**